### ISSUE TITLE
trouble getting simple mocha test to work

### ISSUE THREAD

**Issue by lakamsani** [gosquared/redis-clustr#19]
node 6.9.3 on a Mac.
# package.json
````json
{
  "name": "Test",
  "version": "1.1.2",
  "dependencies": {
     "redis-clustr" : "1.5.2"
   },
  "devDependencies": {
    "mocha": "3.2.0",
     "chai": "3.5.0"
  }
}
````

# simpleTest.js
````json
var expect = require('chai').expect;
var should = require('chai').should()

var RedisClustr = require('redis-clustr');

var redisClient =  new RedisClustr({
  servers: [
    {
      host: 'localhost',
      port: 6379
    }
  ]
})


describe('Redis tests', function () {

  it('should hmset', function (done) {
    redisClient.set('key','value')
    redisClient.hmset("hashmap" , {"name1": "value1", "name2": "value2"})
    redisClient.setex(["hello",30000,'join ended'])
    done();
  })
})
````

# Test execution
````bash
 simple% mocha simpleTest.js


  Redis tests
    ✓ should hmset


  1 passing (14ms)

 simple% redis-cli -h localhost
localhost:6379> get key
(nil)
localhost:6379> get hello
(nil)
localhost:6379> get hashmap
(nil)
localhost:6379>
````

**Comment by jbt** [comment]
Hi @lakamsani is it possible to get any further information on the redis setup you're using (redis version; cluster setup etc)?

One thing to note (and perhaps we should make it clearer in the readme) is that this module is not intended to work with redis _not_ in cluster mode (i.e. a single redis server or basic master-slave replication)

Another thing just from looking at your tests - the `done` function is called without waiting on the response (and possible error) from any of the previous redis commands, which could explain why the test is passing even if it's not behaving correctly.

**Comment by lakamsani** [comment]
@jbt  thanks for that clarification and also for making this library available. I changed the code like this for now (commented out done) and the values are being saved. I was thinking that the sets being async I could just confirm this initial test by looking in the db. But the process is probably existing too soon for the saves to be sent to the cluster (especially if it sends a MOVED for example). I will add separate tests with done being called in the callbacks of the sets. 

Just one further question for you. I 'm trying to use a locking library called [redlock](https://github.com/mike-marcacci/node-redlock). It is not working in the cluster due as it is not handling MOVED errors when I initialize it with a few node_redis client handles. I tried passing a handle to redis-clusr instead and it was stuck on waiting to get a lock. I 'm following up with the author of that package but if you have any pointers on how you are handling MOVED errors  in your library that would be great and I can try and incorporate that into my use of redlock. 

````javascript
var should = require('chai').should()

var redis = require("redis")
var RedisClustr = require('redis-clustr');




var redisClient = new RedisClustr({
  servers: [
    {
      host: 'host1',
      port: 6379
    },
    {
      host: 'host2',
      port: 6379
    }
  ]
})

redisClient.on("error", function (err) {
      console.log("Error " + err);
});

describe('Redis tests', function () {

  it('should test cluster set', function (done) {
    redisClient.set('key','value', redis.print)
    redisClient.hmset("hashmap" , {"name1": "value1", "name2": "value2"}, redis.print)
    redisClient.setex(["hello",30000,'join ended'], redis.print)
    //done();
  })
````
